File: ObjC/AppmediationExample/AppmediationExample/MainScreen.m
//
//  MainScreen.m
//  AppmediationExample
//
//  Created by Mert Celik on 1/14/19.
//  Copyright © 2019 appmediation. All rights reserved.
//

#import "MainScreen.h"
#import "BannerView.h"
#import "CustomBannerView.h"
#import "InterstitialView.h"
#import "RewardedView.h"

@interface MainScreen ()

@end

@implementation MainScreen

- (void)viewDidLoad {
    [super viewDidLoad];
    [self.view setAccessibilityLabel:@"tableView"];
    [self.navigationItem.backBarButtonItem setIsAccessibilityElement:YES];
    [self.navigationItem.backBarButtonItem setAccessibilityLabel:@"Back"];
}

#pragma mark - Table view data source

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    NSLog(@"test");
    switch (indexPath.row) {
        case 0:
            [self navigateTo:Banner];
            break;
        case 1:
            [self navigateTo:CustomBanner];
            break;
        case 2:
            [self navigateTo:Interstitial];
            break;
        case 3:
            [self navigateTo:Rewarded];
            break;
        default:
            //
            break;
    }
}

#pragma mark - Navigation

- (void)navigateTo:(Destination)destination {
    UIViewController *destinationVC;
    switch (destination) {
        case Banner:
            destinationVC = [[BannerView alloc] init];
            break;
        case CustomBanner:
            destinationVC = [[CustomBannerView alloc] init];
            break;
        case Interstitial:
            destinationVC = [[InterstitialView alloc] init];
            break;
        case Rewarded:
            destinationVC = [[RewardedView alloc] init];
            break;
        default:
            destinationVC = nil;
            break;
    }
    [self.navigationController pushViewController:destinationVC animated:YES];
}

@end

File: Swift/AppmediationSwiftExample/AppmediationSwiftExample/MainScreen.swift
//
//  MainScreen.swift
//  AppmediationSwiftExample
//
//  Created by Mert Celik on 1/14/19.
//  Copyright © 2019 appmediation. All rights reserved.
//

import UIKit

class MainScreen: UITableViewController {
    
    enum Destination {
        case Banner
        case CustomBanner
        case Interstitial
        case Rewarded
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.accessibilityLabel = "tableView"
        self.navigationItem.backBarButtonItem?.isAccessibilityElement = true
        self.navigationItem.backBarButtonItem?.accessibilityLabel = "Back"
    }

    // MARK: - Table view data source

    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        switch indexPath.row {
        case 0:
            navigate(to: .Banner)
            break
        case 1:
            navigate(to: .CustomBanner)
            break
        case 2:
            navigate(to: .Interstitial)
            break
        case 3:
            navigate(to: .Rewarded)
            break
        default:
            break
        }
    }
    
    // MARK: Navigation
    
    fileprivate func navigate(to destination: Destination) {
        var destinationVC: UIViewController!
        switch destination {
        case .Banner:
            destinationVC = BannerView()
            break
        case .CustomBanner:
            destinationVC = CustomBannerView()
            break
        case .Interstitial:
            destinationVC = InterstitialView()
            break
        case .Rewarded:
            destinationVC = RewardedView()
            break
        }
        self.navigationController?.pushViewController(destinationVC, animated: true)
    }

}
